avocado/utils/network/interfaces.py

```diff
@@ -152,6 +164,9 @@ def is_link_up(self):
         """
         if self.get_link_state() == 'up':
             return True
+        elif self.get_link_state() == 'unknown':
+            if 'LOWER_UP' in self.get_admin_link_state():
+                return True
```

Comment:
Here you are changing the logic of this method. Not sure if we need this change here. Please see my first comment here.

Reply:
@beraldoleal actually the problem we are facing with veth interface (virtual interface), it is in unknown state, so for this we wanted to check with operational state,is the interface is  up or down through "LOWER_UP"
```
env3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UNKNOWN
```

Reply:
Again, IMO this is changing the behavior here.

Here is what I'm thinking: With your change, adding the method `get_operational_link_state()`, you have introduced the separation between physical (operational) and logical (administrative) checks here, and this **is very nice**.

But we should keep this separation on the other methods: `is_link_up()` could be a method to check if the link is "fully UP" (operational *and* administrative), because there is no admin neither operational in its name.

Imagine this scenario: Link is physically UP (operational) but administratively down. With your change, this will return `True`. Do you see it?

We need to decide what `is_link_up()` will check. IMO we should check for a *AND* on both: administrative and operational.

What do you think?

Reply:
@beraldoleal yes we can do that,i about to add that.